src/main/java/org/myrobotlab/net/Http.java

```diff
@@ -117,5 +124,17 @@ public static void get(String theUrl, String outFile) throws IOException {
       in.close(); // call this in a finally block
       out.close();
   }
+  
+  public static void getFile(String url) throws IOException {
+    getFile(url, null);
+  }
+
+  public static void getFile(String url, String outFile) throws IOException {
+    if (outFile == null) {
+      outFile = url.substring(url.lastIndexOf("/") + 1);

```

Comment:
this url parsing is not actually correct..  it should be the last index of /  before the first index of ? or a #    (if a question mark or hash tag are in the url.)   Probably best to actually create a URL object in java.. to parse the url as a proper url and get the path...

Reply:
people don't supply urls - they supply strings .. so there must be something which accepts a string even if it does get turned into a url .. I'm fine with the URL parameter being the function of "maximum complexity" which all other overloads just do type conversions to finally get to that method with the URL ...   I'd not necessarily agree with your ? or # statement however because a file can be the "result" of a query to some cgi or angular function - I think the thing I needed at the time was the outFile

Reply:
Added to my short list "Http.getFile use URL/URI parsing" .. but low priority at the moment